conformance/lang/expressions/logical-expressions/logical_and.balt

```diff
@@ -0,0 +1,732 @@
+    boolean _ =  1-2 && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  1+2 && 1-2; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using shift expression.
+Labels: logical-expr, shift-expr
+
+function init() {
+    boolean _ =  true && 1 << 2; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  1 >> 2 && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  1 >> 2 && 1 << 2; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using range expression.
+Labels: logical-expr, range-expr
+
+function init() {
+    boolean _ =  true && (1 ..< 5).iterator().next(); // @error static type of each expression must be a subtype of boolean
+    boolean _ =  (1 ... 5).iterator().next() && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  (1 ..< 5).iterator().next() && (1 ... 5).iterator().next(); // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using binary bitwise expression.
+Labels: logical-expr, binary-bitwise-expr
+
+function init() {
+    boolean _ =  true && 97 & 64; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  97 & 64 && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  97 & 64 && 97 & 64; // @error static type of each expression must be a subtype of boolean
+
+    boolean _ =  true && 97 ^ 64; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  97 ^ 64 && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  97 ^ 64 && 97 ^ 64; // @error static type of each expression must be a subtype of boolean
+
+    boolean _ =  true && 97 | 64; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  97 | 64 && true; // @error static type of each expression must be a subtype of boolean
+    boolean _ =  97 | 64 && 97 | 64; // @error static type of each expression must be a subtype of boolean
+}
+
+Test-Case: error
+Description: Test static type of expression in logical expression via invalid types using conditional expression.
+Fail-issue: ballerina-lang #33358
+Labels: logical-expr, conditional-expr
+
+function init() {
+    string a = "text";
+    io:println(true && a == "text"? 2 : 1); // @error static type of each expression must be a subtype of boolean
+    io:println(false && a == "text"? 2 : 1); // @error static type of each expression must be a subtype of boolean
```

Comment:
We don't use `io:println` in `error` tests. Please rewrite such tests without `io:println`.